<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1080</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QPushButton" name="uploadfile">
      <property name="text">
       <string>upload file</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Original imamge</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="originalimage"/>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>kernel size (1~99)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>spatial filtering type</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="spatialFilter_kernelSize_spinbox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>99</number>
        </property>
        <property name="singleStep">
         <number>2</number>
        </property>
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="spatialFilter_comboBox">
        <item>
         <property name="text">
          <string>median 中位</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>mean 平均</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>max 最大</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>min 最小</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="ApplySpatialFilter">
        <property name="text">
         <string>ApplySpatialFilter</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Spatial Filter</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="spatialFilterImage"/>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>gauss kernel size 
(1~99)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>zero cross 
threshold (0.0~99.0)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="log_kernelSize_spinbox">
        <property name="maximum">
         <number>99</number>
        </property>
        <property name="singleStep">
         <number>2</number>
        </property>
        <property name="value">
         <number>21</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>gauss sigma 
(0.01~5.0)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="log_sigma_spinbox">
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>5.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>2.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="ApplySobelandLoG">
        <property name="text">
         <string>Apply sobel and LoG</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="log_threshold_spinbox"/>
      </item>
     </layout>
    </item>
    <item row="2" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>sobel</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="sobelImage"/>
      </item>
     </layout>
    </item>
    <item row="2" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>LoG</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="marrHildrethImage"/>
      </item>
     </layout>
    </item>
    <item row="3" column="0">
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>local kernel size 
(1~99)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>k0 (0.0~5.0)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>k1 (0.0~5.0)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>k2 (0.0~5.0)</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>k3 (0.0~5.0)</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>C (0.0~999.0)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="localStatisticalEnhancement_kernelSize_spinbox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>99</number>
        </property>
        <property name="singleStep">
         <number>2</number>
        </property>
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="localStatisticalEnhancement_k0_spinbox">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>5.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="localStatisticalEnhancement_k1_spinbox">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>5.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.250000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="localStatisticalEnhancement_k2_spinbox">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>5.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="localStatisticalEnhancement_k3_spinbox">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>5.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QDoubleSpinBox" name="localStatisticalEnhancement_C_spinbox">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>999.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>22.800000000000001</double>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QPushButton" name="ApplylocalMeanConstrastEnhancement">
        <property name="text">
         <string>Apply Local Mean Contrast Enhancement</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="3" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>equalize Histogram </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="equalizeHistogramRGBImage"/>
      </item>
     </layout>
    </item>
    <item row="3" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>localMeanContrastEnhancement</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="localMeanContrastEnhancementImage"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1080</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
